drivers/regulator/regulator_nxp_vref.c

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2023 NXP
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_vref
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/dt-bindings/regulator/nxp_vref.h>
+#include <errno.h>
+#include <zephyr/sys/util.h>
+
+#define NXP_VREF_MIN_UV 1000000
+#define NXP_VREF_MAX_UV 2100000
+
+#define NXP_VREF_UV_PER_TRIM2V1_UNIT 1000000
+
+struct nxp_vref_vmap {
+	uint32_t voltage;
+	uint8_t reg_value;
+};
+
+static const struct nxp_vref_vmap utrim_vals[] = {
+	{ 1000000, 0b0000 },
+	{ 1100000, 0b0001 },
+	{ 1200000, 0b0010 },
+	{ 1300000, 0b0011 },
+	{ 1400000, 0b0100 },
+	{ 1500000, 0b0101 },
+	{ 1600000, 0b0110 },
+	{ 1700000, 0b0111 },
+	{ 1800000, 0b1000 },
+	{ 1900000, 0b1001 },
+	{ 2000000, 0b1010 },
+	{ 2100000, 0b1011 },
+};
```

Comment:
okay, I can convert to this if desired

Reply:
If you convert this to a linear range it will simplify the checking further down.

```suggestion
static const struct linear_range vref_range = LINEAR_RANGE_INIT(1000000, 100000U, 0x0U, 0xBU);
```